### PR TITLE
[FIX]: fix DpasLayout offset calculate

### PR DESCRIPTION
**Pull request by quintinwang5**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 1]
```diff
@@ -887,8 +887,7 @@ static void emitDpasOffsetForCTA(const DpasEncodingAttr &dpasLayout,
   for (unsigned elem = 0; elem < elemsPerThreadPerGroup; elem++) {
     uint32_t elemRowIndex = (elem / sizePerThreads[1]) * rowsPerWarp;
     uint32_t elemColIndex = elem % sizePerThreads[1];
-    offsets.push_back({ctaOffsetX * shapePerCTA[0] + elemRowIndex,
-                       ctaOffsetY * shapePerCTA[1] + elemColIndex});
+    offsets.push_back({ctaOffsetX + elemRowIndex, ctaOffsetY + elemColIndex});
```
[the remark this answers is not in the document]
Comment (quintinwang5) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:890]: https://github.com/intel/intel-xpu-backend-for-triton/blob/7810347e9fb36b8f0630590688452b5d50b2de41/third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h#L980 Seems it's already embedded here by i/j.

[the remark this answers is not in the document]
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:890]: Yeah, you are right.
It seems the stride is duplicated accounted.

The better to align to the offset calculation utils function to the NV. Like: `emitOffsetForMmaLayoutV3`. 
Inplace the `emitDpasOffsetForCTA` to the `emitOffsetForDpasLayout`. 

The name of `emitDpasOffsetForCTA` is confusing.

[the remark this answers is not in the document]
Comment (quintinwang5) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:891]: Refined. As it's just a function used in inner loop of `emitOffsetForDpasLayout` per CTA, change it to `emitOffsetForDpasLayoutPerCTA`

**Review by etiotto — APPROVED** [review]

**Comment by chengjunlu** [comment]
The fix LGTM. 
Note: It is a warp stride for the replica when the size of `getShapePerCTATile` is smaller to the shape of tensor.

We will revisit it when we use a dense stride for the replica later for large 2D block load/store.

**Review by chengjunlu — APPROVED** [review]

**Pull request merged**